src/master-playlist-controller.js

```diff
@@ -957,55 +961,79 @@ export class MasterPlaylistController extends videojs.EventTarget {
       return;
     }
 
-    this.pauseLoading();
     const logFn = error.internal ? this.logger_ : videojs.log.warn;
     const errorMessage = error.message ? (' ' + error.message) : '';
 
     logFn(`${(error.internal ? 'Internal problem' : 'Problem')} encountered with playlist ${currentPlaylist.id}.` +
       `${errorMessage} Switching to playlist ${nextPlaylist.id}.`);
 
+    // if audio group changed reset audio loaders
+    if (nextPlaylist.attributes.AUDIO !== currentPlaylist.attributes.AUDIO) {
+      this.delegateLoaders_('audio', ['abort', 'pause']);
+    }
+
+    // if subtitle group changed reset subtitle loaders
+    if (nextPlaylist.attributes.SUBTITLES !== currentPlaylist.attributes.SUBTITLES) {
+      this.delegateLoaders_('subtitle', ['abort', 'pause']);
+    }
+
+    this.delegateLoaders_('main', ['abort', 'pause']);
+
     return this.masterPlaylistLoader_.media(nextPlaylist, isFinalRendition);
   }
 
   /**
    * Pause all segment/playlist loaders
    */
   pauseLoading() {
-    this.delegateLoaders_(['abort', 'pause']);
+    this.delegateLoaders_('all', ['abort', 'pause']);
   }
 
   /**
    * Call a set of functions in order on playlist loaders, segment loaders,
    * or both types of loaders.
    *
+   * @param {string} filter
+   *        Filter loaders that should call fnNames using a string. Can be:
+   *        * all - run on all loaders
+   *        * playlist - run on all playlist loaders
+   *        * segment - run on all segment loaders
+   *        * audio - run on all audio loaders
+   *        * subtitle - run on all subtitle loaders
+   *        * video/main - run on the main/master loaders
```

Comment:
Is it worth having video as an option if we only use this internally?

Reply:
probably not.